### ISSUE TITLE
"Unix.Unix_error "No such file or directory" open" during scrubbing

### ISSUE THREAD

**Issue by kinvaris** [openvstorage/volumedriver#116]
### On master node: ctrl-roub-01:
```
2016-09-19 11:58:00 62600 +0200 - ctrl-roub-01 - 93351/140310253107008 - lib/scheduled tasks - 28 - ERROR - Gather Scrub - Storage Router 172.20.20.31    - Scrubbing failed with error:
 - Failed scrubbing work unit for volume KVM01-fio-rou-d1 with guid 7c1c1d9e-6453-4a97-9a5d-8b524de1f2ac: scrubber returned with nonzero exit status 256
```
### On node in question: perf-roub-01:
```
2016-09-19 11:56:56 55100 +0200 - perf-roub-01 - 62208/139790198290240 - celery/celery.redirected - 815 - WARNING - 2016-09-19 11:56:56 55100 +0200 - perf-roub-01 - 62208/139790198290240 - extensions/volatile mutex - 814 - WARNING - Waited 0.103229045868 sec for lock ovs_lock_messaging
2016-09-19 11:56:56 60300 +0200 - perf-roub-01 - 62180/139790198290240 - celery/celery.worker.job - 16320 - ERROR - Task ovs.scheduled.execute_scrub_work[5f0fee95-d6cf-40fe-8bbc-2f243da17736] raised unexpected: Exception('Failed scrubbing work unit for volume KVM01-fio-rou-d1 with guid 7c1c1d9e-6453-4a97-9a5d-8b524de1f2ac: scrubber returned with nonzero exit status 256',)
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "/opt/OpenvStorage/ovs/lib/scheduledtask.py", line 376, in _execute_scrub_work
    raise Exception('\n - '.join(failures))
Exception: Failed scrubbing work unit for volume KVM01-fio-rou-d1 with guid 7c1c1d9e-6453-4a97-9a5d-8b524de1f2ac: scrubber returned with nonzero exit status 256
```
### Full task on perf-roub-01:
```
2016-09-19 11:56:56 16700 +0200 - perf-roub-01 - 62180/139790198290240 - celery/celery.worker.job - 16319 - DEBUG - Task accepted: ovs.scheduled.execute_scrub_work[5f0fee95-d6cf-40fe-8bbc-2f243da17736] pid:62208
2016-09-19 11:56:56 16900 +0200 - perf-roub-01 - 62208/139790198290240 - lib/scheduled tasks - 809 - INFO - Execute Scrub - Started
2016-09-19 11:56:56 17000 +0200 - perf-roub-01 - 62208/139790198290240 - lib/scheduled tasks - 810 - INFO - Execute Scrub - Scrub location - /mnt/hdd3/geo-accellerated-alba_scrub_1
2016-09-19 11:56:56 18400 +0200 - perf-roub-01 - 62208/139790198290240 - lib/scheduled tasks - 811 - INFO - Execute Scrub - Virtual disk 7c1c1d9e-6453-4a97-9a5d-8b524de1f2ac - KVM01-fio-rou-d1 - Started
2016-09-19 11:56:56 37000 +0200 - perf-roub-01 - 62208/139790198290240 - lib/scheduled tasks - 812 - INFO - Execute Scrub - Virtual disk 7c1c1d9e-6453-4a97-9a5d-8b524de1f2ac - KVM01-fio-rou-d1 - Retrieve and apply scrub work
libust[48335/48335]: Warning: HOME environment variable not set. Disabling LTTng-UST per-user tracing. (in setup_local_apps() at lttng-ust-comm.c:305)
2016-09-19 11:56:56 412076 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/ovs_scrubber - 0000000000000000 - notice - print_version_data: version: 6.0.4
version revision: cf03dad081dcc113f6121a6c912c8a755d8949d6
branch: dev
md5 support: no
revision: 7a8da553c2d2cfc98d25c33c38306f2911ed72cc
build time: Tue Sep  6 00:33:42 UTC 2016

2016-09-19 11:56:56 412377 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/ovs_scrubber - 0000000000000001 - notice - print_version_data: Command Line Options
2016-09-19 11:56:56 413118 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/Scrubber - 0000000000000002 - info - operator(): Instantiation the FilePool
2016-09-19 11:56:56 413202 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/FilePool - 0000000000000003 - info - FilePool: Creating new FilePool at /mnt/hdd3/geo-accellerated-alba_scrub_1/dced2004-fa09-4bd5-8fe9-db9dff8d7
5db
2016-09-19 11:56:56 413587 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/Scrubber - 0000000000000004 - info - operator(): getting the sco access data from the backend
2016-09-19 11:56:56 413860 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/AlbaConnection - 0000000000000005 - info - Connection: Setting up Alba log handler
2016-09-19 11:56:56 413979 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/AlbaProxyClient - 0000000000000006 - info - logMessage: TCPProxy_client(172.20.20.33, 26204)
2016-09-19 11:56:56 414416 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/BackendConnectionInterfaceLogger - 0000000000000007 - info - Logger: Entering read 1530ce26-25a4-4ce1-adcb-5ee245cb9486 sco_access_data
2016-09-19 11:56:56 423739 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/BackendConnectionInterfaceLogger - 0000000000000008 - info - ~Logger: Exiting read for 1530ce26-25a4-4ce1-adcb-5ee245cb9486 sco_access_data
2016-09-19 11:56:56 423806 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/BackendConnectionInterfaceLogger - 0000000000000009 - info - ~Logger: Exiting read for 1530ce26-25a4-4ce1-adcb-5ee245cb9486
2016-09-19 11:56:56 424154 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/NormalizedSCOAccessData - 000000000000000a - warning - NormalizedSCOAccessData: No SCO Access Data for this scrubbing run found
2016-09-19 11:56:56 424214 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/Scrubber - 000000000000000b - info - operator(): Getting the snapshots file from the backend
2016-09-19 11:56:56 424292 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/BackendConnectionInterfaceLogger - 000000000000000c - info - Logger: Entering read 1530ce26-25a4-4ce1-adcb-5ee245cb9486 snapshots.xml
2016-09-19 11:56:56 427888 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/AlbaConnection - 000000000000000d - error - convert_exceptions_: read object: caught Alba proxy exception: (Unix.Unix_error "No such file or directory" open
  /mnt/hdd3/geo-accellerated-alba_scrub_1/dced2004-fa09-4bd5-8fe9-db9dff8d75db/snapshotsAbzdtA)
2016-09-19 11:56:56 428153 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/BackendConnectionInterfaceLogger - 000000000000000e - error - ~Logger: Exiting read for 1530ce26-25a4-4ce1-adcb-5ee245cb9486 snapshots.xml with exception
2016-09-19 11:56:56 428231 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/BackendConnectionInterfaceLogger - 000000000000000f - error - ~Logger: Exiting read for 1530ce26-25a4-4ce1-adcb-5ee245cb9486
2016-09-19 11:56:56 428300 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/BackendInterface - 0000000000000010 - error - do_wrap_: Problem with connection 0x11220d0: (Unix.Unix_error "No such file or directory" open
  /mnt/hdd3/geo-accellerated-alba_scrub_1/dced2004-fa09-4bd5-8fe9-db9dff8d75db/snapshotsAbzdtA)
2016-09-19 11:56:56 428515 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/AlbaProxyClient - 0000000000000011 - info - logMessage: TCPProxy_client(172.20.20.33, 26204)
2016-09-19 11:56:56 428882 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/BackendInterface - 0000000000000012 - warning - do_wrap_: Retrying with new connection (retry: 1, sleep before retry: 0 milliseconds)
2016-09-19 11:56:56 429071 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/BackendConnectionInterfaceLogger - 0000000000000013 - info - Logger: Entering read 1530ce26-25a4-4ce1-adcb-5ee245cb9486 snapshots.xml
2016-09-19 11:56:56 431327 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/AlbaConnection - 0000000000000014 - error - convert_exceptions_: read object: caught Alba proxy exception: (Unix.Unix_error "No such file or directory" open
  /mnt/hdd3/geo-accellerated-alba_scrub_1/dced2004-fa09-4bd5-8fe9-db9dff8d75db/snapshotsDDCFQz)
2016-09-19 11:56:56 428153 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/BackendConnectionInterfaceLogger - 000000000000000e - error - ~Logger: Exiting read for 1530ce26-25a4-4ce1-adcb-5ee245cb9486 snapshots.xml with exception
2016-09-19 11:56:56 428231 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/BackendConnectionInterfaceLogger - 000000000000000f - error - ~Logger: Exiting read for 1530ce26-25a4-4ce1-adcb-5ee245cb9486
2016-09-19 11:56:56 428300 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/BackendInterface - 0000000000000010 - error - do_wrap_: Problem with connection 0x11220d0: (Unix.Unix_error "No such file or directory" open
  /mnt/hdd3/geo-accellerated-alba_scrub_1/dced2004-fa09-4bd5-8fe9-db9dff8d75db/snapshotsAbzdtA)
2016-09-19 11:56:56 428515 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/AlbaProxyClient - 0000000000000011 - info - logMessage: TCPProxy_client(172.20.20.33, 26204)
2016-09-19 11:56:56 428882 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/BackendInterface - 0000000000000012 - warning - do_wrap_: Retrying with new connection (retry: 1, sleep before retry: 0 milliseconds)
2016-09-19 11:56:56 429071 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/BackendConnectionInterfaceLogger - 0000000000000013 - info - Logger: Entering read 1530ce26-25a4-4ce1-adcb-5ee245cb9486 snapshots.xml
2016-09-19 11:56:56 431327 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/AlbaConnection - 0000000000000014 - error - convert_exceptions_: read object: caught Alba proxy exception: (Unix.Unix_error "No such file or directory" open
  /mnt/hdd3/geo-accellerated-alba_scrub_1/dced2004-fa09-4bd5-8fe9-db9dff8d75db/snapshotsDDCFQz)
2016-09-19 11:56:56 431505 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/BackendConnectionInterfaceLogger - 0000000000000015 - error - ~Logger: Exiting read for 1530ce26-25a4-4ce1-adcb-5ee245cb9486 snapshots.xml with exception
2016-09-19 11:56:56 431577 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/BackendConnectionInterfaceLogger - 0000000000000016 - error - ~Logger: Exiting read for 1530ce26-25a4-4ce1-adcb-5ee245cb9486
2016-09-19 11:56:56 431635 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/BackendInterface - 0000000000000017 - error - do_wrap_: Problem with connection 0x11220d0: (Unix.Unix_error "No such file or directory" open
  /mnt/hdd3/geo-accellerated-alba_scrub_1/dced2004-fa09-4bd5-8fe9-db9dff8d75db/snapshotsDDCFQz)
2016-09-19 11:56:56 431693 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/BackendInterface - 0000000000000018 - error - do_wrap_: Giving up connection 0x11220d0
2016-09-19 11:56:56 431925 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/FilePool - 0000000000000019 - info - ~FilePool: Cleaning up FilePool at /mnt/hdd3/geo-accellerated-alba_scrub_1/dced2004-fa09-4bd5-8fe9-db9dff8d75db
2016-09-19 11:56:56 432360 +0200 - perf-roub-01 - 48335/0x00007fc7c8625700 - scrubber/SignalThread - 000000000000001a - info - run_: exiting signal handler thread
2016-09-19 11:56:56 432594 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/ovs_scrubber - 000000000000001b - fatal - operator(): caught exception: Exception while running the scrubber: (Unix.Unix_error "No such file or directory" open
  /mnt/hdd3/geo-accellerated-alba_scrub_1/dced2004-fa09-4bd5-8fe9-db9dff8d75db/snapshotsDDCFQz)
2016-09-19 11:56:56 432704 +0200 - perf-roub-01 - 48335/0x00007fc7d2692940 - scrubber/ovs_scrubber - 000000000000001c - notice - at_exit: Time spent executing user instructions (seconds): 0.8000
Time spent in operating system code (seconds): 0.24000
The maximum resident set size used, in kilobytes. 78732
The number of page faults which were serviced without requiring any I/O: 988
The number of page faults which were serviced by doing I/O: 0
The number of times PROCESSES was swapped entirely out of main memory: 0
The number of times the file system had to read from the disk: 0
The number of times the file system had to write to the disk: 0
The number of signals received: 0
The number of times PROCESSES voluntarily invoked a context switch: 10
The number of times an involuntary context switch took place: 80
2016-09-19 11:56:56 44300 +0200 - perf-roub-01 - 62208/139790198290240 - lib/scheduled tasks - 813 - INFO - Execute Scrub - Finished - Success: 0 - Failed: 1 - Skipped: 0
2016-09-19 11:56:56 55100 +0200 - perf-roub-01 - 62208/139790198290240 - celery/celery.redirected - 815 - WARNING - 2016-09-19 11:56:56 55100 +0200 - perf-roub-01 - 62208/139790198290240 - extensions/volatile mutex - 814 - WARNING - Waited 0.103229045868 sec for lock ovs_lock_messaging
2016-09-19 11:56:56 60300 +0200 - perf-roub-01 - 62180/139790198290240 - celery/celery.worker.job - 16320 - ERROR - Task ovs.scheduled.execute_scrub_work[5f0fee95-d6cf-40fe-8bbc-2f243da17736] raised unexpected: Exception('Failed scrubbing work unit for volume KVM01-fio-rou-d1 with guid 7c1c1d9e-6453-4a97-9a5d-8b524de1f2ac: scrubber returned with nonzero exit status 256',)
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "/opt/OpenvStorage/ovs/lib/scheduledtask.py", line 376, in _execute_scrub_work
    raise Exception('\n - '.join(failures))
Exception: Failed scrubbing work unit for volume KVM01-fio-rou-d1 with guid 7c1c1d9e-6453-4a97-9a5d-8b524de1f2ac: scrubber returned with nonzero exit status 256
```
Volumedriver log `/var/log/upstart/ovs-volumedriver_geo-accellerated-alba.log`:
```
2016-09-19 11:56:56 275280
...
2016-09-19 11:57:53 424085
```

**Comment by redlicha** [comment]
Log inspection (`/var/log/ovs/storagerouterclient.log`) suggests that the scrubber is pointed to the wrong proxy:
```
2016-09-19 11:56:56 379616 +0200 - perf-roub-01 - 62208/0x00007f23711cb740 -LoggerToolCut/LockedPythonClient - 0000000000000065 - info - scrub: 1530ce26-25a4-4ce1-adcb-5ee245cb9486: launching scrubber, args: 
2016-09-19 11:56:56 379639 +0200 - perf-roub-01 - 62208/0x00007f23711cb740 - LoggerToolCut/LockedPythonClient - 0000000000000066 - info - scrub:        ovs_scrubber
2016-09-19 11:56:56 379656 +0200 - perf-roub-01 - 62208/0x00007f23711cb740 -LoggerToolCut/LockedPythonClient - 0000000000000067 - info - scrub:        --scratch-dir
2016-09-19 11:56:56 379670 +0200 - perf-roub-01 - 62208/0x00007f23711cb740 -LoggerToolCut/LockedPythonClient - 0000000000000068 - info - scrub:        /mnt/hdd3/geo-accellerated-alba_scrub_1
2016-09-19 11:56:56 379685 +0200 - perf-roub-01 - 62208/0x00007f23711cb740 -LoggerToolCut/LockedPythonClient - 0000000000000069 - info - scrub:        --region-size-exponent
2016-09-19 11:56:56 379699 +0200 - perf-roub-01 - 62208/0x00007f23711cb740 -LoggerToolCut/LockedPythonClient - 000000000000006a - info - scrub:        25
2016-09-19 11:56:56 379732 +0200 - perf-roub-01 - 62208/0x00007f23711cb740 -LoggerToolCut/LockedPythonClient - 000000000000006b - info - scrub:        --fill-ratio
2016-09-19 11:56:56 379749 +0200 - perf-roub-01 - 62208/0x00007f23711cb740 -LoggerToolCut/LockedPythonClient - 000000000000006c - info - scrub:        0.899999976
2016-09-19 11:56:56 379763 +0200 - perf-roub-01 - 62208/0x00007f23711cb740 -LoggerToolCut/LockedPythonClient - 000000000000006d - info - scrub:        --scrub-work
2016-09-19 11:56:56 379777 +0200 - perf-roub-01 - 62208/0x00007f23711cb740 -LoggerToolCut/LockedPythonClient - 000000000000006e - info - scrub:        <?xml version="1.0" encoding="UTF-8" standalone="yes" ?>
<!DOCTYPE boost_serialization>
<boost_serialization signature="serialization::archive" version="11">
<scrubwork class_id="0" tracking_level="0" version="2">
[...]
</scrubwork>

2016-09-19 11:56:56 379817 +0200 - perf-roub-01 - 62208/0x00007f23711cb740 -LoggerToolCut/LockedPythonClient - 000000000000006f - info - scrub:        --verbose
2016-09-19 11:56:56 379833 +0200 - perf-roub-01 - 62208/0x00007f23711cb740 -LoggerToolCut/LockedPythonClient - 0000000000000070 - info - scrub:        1
2016-09-19 11:56:56 379847 +0200 - perf-roub-01 - 62208/0x00007f23711cb740 -LoggerToolCut/LockedPythonClient - 0000000000000071 - info - scrub:        --loglevel
2016-09-19 11:56:56 379861 +0200 - perf-roub-01 - 62208/0x00007f23711cb740 -LoggerToolCut/LockedPythonClient - 0000000000000072 - info - scrub:        info
2016-09-19 11:56:56 379875 +0200 - perf-roub-01 - 62208/0x00007f23711cb740 -LoggerToolCut/LockedPythonClient - 0000000000000073 - info - scrub:        --logsink
2016-09-19 11:56:56 379888 +0200 - perf-roub-01 - 62208/0x00007f23711cb740 -LoggerToolCut/LockedPythonClient - 0000000000000074 - info - scrub:        console:
2016-09-19 11:56:56 436003 +0200 - perf-roub-01 - 62208/0x00007f23711cb740 -LoggerToolCut/LockedPythonClient - 0000000000000075 - info - scrub: 1530ce26-25a4-4ce1-adcb-5ee245cb9486: scrubber said: 
2016-09-19 11:56:56 436138 +0200 - perf-roub-01 - 62208/0x00007f23711cb740 -LoggerToolCut/LockedPythonClient - 0000000000000076 - error - scrub: 1530ce26-25a4-4ce1-adcb-5ee245cb9486: scrubber's exit status: 256
```

IOW the commandline of the scrubber *does not* contain a `--backend-config <cfg>` ( cf. #53 ), as the framework side of this is not yet implemented - cf. https://github.com/openvstorage/framework/issues/72/ .

@khenderick / @wimpers: unless you want to move this over to the framework tracker I'm going to close this.

**Comment by redlicha** [comment]
Let me rephrase: I'm closing this here, please move over to the framework / reopen there if required. :)

**Comment by wimpers** [comment]
@khenderick  do we need to open it in the framework repo?

**Comment by khenderick** [comment]
No, an integration ticket already exists. See in ine of the first comments.
